python/pyarrow/_compute.pxd

```diff
@@ -21,11 +21,11 @@ from pyarrow.lib cimport *
 from pyarrow.includes.common cimport *
 from pyarrow.includes.libarrow cimport *
 
-cdef class ScalarUdfContext(_Weakrefable):
+cdef class UdfContext(_Weakrefable):
```

Comment:
For the moment, let's leave this as `ScalarUdfContext`. The reason is not that you are wrong, for the moment the version we have only supports Scalr UDFs and probably Tabular UDFs with your PR. But the aggregate UDFs are work in progress (there is a PR for that too https://github.com/apache/arrow/pull/14527). Once we finalize this work, then let's see if we want to extend from `UDFContext` and write the others depending on the way the experimental API would evolve.

Reply:
There is another issue, that I recall that I didn't name it like this. `CUdf` can get mixed up with `CudfContext` or related things. Need to come up with better naming too.

Reply:
And probably if we are to do a name change for a specific reason, shall we do it in a separate PR and just keep the feature introduced to this PR?

Reply:
I don't mind undoing this name change. IIRC, @westonpace requested it, so he should have an opinion.

As a side note, I wouldn't generally make an effort just to avoid a conflict with another pending PR. If a PR is a step in the right direction, then this is good enough, and any conflict will be resolved in remaining PRs.

Reply:
> There is another issue, that I recall that I didn't name it like this. `CUdf` can get mixed up with `CudfContext` or related things. Need to come up with better naming too.

I don't follow. At least in the recent commit, [`UdfContext` is a wrapper of `CUdfContext`](https://github.com/apache/arrow/blob/4236e37aaeef7fb92de4b241f3202f4e8a40463d/python/pyarrow/_compute.pyx#L2522-L2535). Where's the mix up?

Reply:
I think I didn't put it accurately, I am not worried about the conflicts or rebasing the changes, but constant renaming in each PR until we figure things out.

Reply:
I [reverted](https://github.com/apache/arrow/pull/14682/commits/8a4d820131ba919147a0704ff91acf100c745384) the naming changes.